guide.md

```diff
@@ -0,0 +1,350 @@
+# A guide to rules_nixpkgs
+
+TODO: does use flake work with vanilla direnv? nix-direnv is just an improvement over that, i believe
```

Comment:
Thanks, I've added the `source_url` directive and added a note about the versions. Since nix-direnv significantly improves upon the support present in vanilla direnv in ways users are likely to notice very soon (caching and automatic GC roots), it makes sense to start users off with it in my opinion.